calico_node/startup/startup.go

```diff
@@ -267,7 +274,11 @@ func configureIPsAndSubnets(node *api.Node) {
 		adm := os.Getenv("IP_AUTODETECTION_METHOD")
 		cidr := autoDetectCIDR(adm, 4)
 		if cidr != nil {
-			// We autodetected an IPv4 address so update the value in the node.
+			// We autodetected an IPv4 address, if we have no previous IP, or it is different than the previous IP,
```

Comment:
I think this comment is wrong now that we've moved the check outside of this branch - should probably just be reverted to what it was.